lib/ansible/modules/cloud/azure/azure_rm_snapshot.py

```diff
@@ -0,0 +1,378 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Zim Kalinowski, (@zikalino)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_snapshot
+version_added: '2.9'
+short_description: Manage Azure Snapshot instance.
+description:
+  - 'Create, update and delete instance of Azure Snapshot.'
+options:
+  resource_group:
+    description:
+      - The name of the resource group.
+    required: true
+    type: str
+  name:
+    description:
+      - Resource name
+    type: str
+  location:
+    description:
+      - Resource location
+    required: true
```

Comment:
is location always required even when updating?

Reply:
fixed, not required